arch/riscv/core/coredump.c

```diff
@@ -14,6 +14,8 @@
 #define ARCH_HDR_VER 2
 #endif
```

Comment:
When changing the arch block, you should adjust the header version to maintain backwards compatibility.

It looks like the version is currently being used to differentiate 32 vs. 64 bit unfortunately. This information can be determined elsewhere in the dump, checking the "Pointer size" in the file header, not sure if it's exposed though.

Reply:
We can read the pointer size from the coredump to differentiate between 32 vs 64 bit, but we still have RV32E which has a smaller set of registers and isn't supported by the coredump parser. Instead of relying on `ARCH_HDR_VER`, I guess we should have created another field in the struct to differentiate between the CPUs (similar to xtensa), but that can be in another PR.